### ISSUE TITLE
[Question] Log message unblocked threads in `_queueSemaphore`

### ISSUE THREAD

**Issue by kinarobin** [CocoaLumberjack/CocoaLumberjack#1189]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](https://github.com/CocoaLumberjack/CocoaLumberjack#documentation) or [CocoaDocs](cocoadocs.org/docsets/CocoaLumberjack)
* [x] I have searched for a similar issue in the [project](https://github.com/CocoaLumberjack/CocoaLumberjack/issues) and found none

### Question Info

From source code:

```objc
- (void)queueLogMessage:(DDLogMessage *)logMessage asynchronously:(BOOL)asyncFlag {
 
    dispatch_block_t logBlock = ^{
        dispatch_semaphore_wait(_queueSemaphore, DISPATCH_TIME_FOREVER);
        @autoreleasepool {
            [self lt_log:logMessage];
        }
    };

    if (asyncFlag) {
        dispatch_async(_loggingQueue, logBlock);
    } else if (dispatch_get_specific(GlobalLoggingQueueIdentityKey)) {
        logBlock();
    } else {
        dispatch_sync(_loggingQueue, logBlock);
    }
}

- (void)lt_log:(DDLogMessage *)logMessage {
    for (DDLoggerNode *loggerNode in self._loggers) {
        if (!(logMessage->_flag & loggerNode->_level)) {
            continue;
        }
        dispatch_group_async(_loggingGroup, loggerNode->_loggerQueue, ^{
            @autoreleasepool {
                [loggerNode->_logger logMessage:logMessage];
            }
        });
    }
    dispatch_group_wait(_loggingGroup, DISPATCH_TIME_FOREVER);
    dispatch_semaphore_signal(_queueSemaphore);
}

```

In default,  the static `_loggingQueue` is `SERIAL` queue, It means every `lt_ log:` methods are executed synchronously.
And in `lt_log:` method internal, the loggingGroup wait logMessage to execute finished. So in DDLog internal, even if `asynchronously` equals YES, each log is executed in FIFO.

My question:

There is no need to use `_queueSemaphore` ? because the thread(in _loggingQueue) is not blocked at all!

Please forgive me for my rudimentary knowledge.

**Comment by ffried** [comment]
You're right that this looks totally strange. Also, the corresponding comments don't help much (in the current situation), but they did give me a hint.
The semaphore _used to be_ waited for outside of the `_loggingQueue`. [This change](https://github.com/CocoaLumberjack/CocoaLumberjack/pull/911/files#diff-5b149257bf8630ed7ccb2b4156146e0a287798b858a6d95040eab1b78bb6a1bdL354) moved it into the serial queue.

While this made the semaphore useless for _most_ situations, there's still the case of internally queued log messages that might overflow the queue. However, this is currently not the case.
I wonder if we should remove the queue-limiting feature altogether. Given that we had no complaints about it no longer working since #911 rendered it nonfunctional, this seems like a reasonable thing to do.

**Comment by kinarobin** [comment]
From #911 fix, some user still had the deadlock issue. If you think that `_queueSemaphore ` doesn't make much sense, I'm glad to submit a PR to remove the `_queueSemaphore `.
